src/audio/codec_adapter/codec_adapter.c

```diff
@@ -395,9 +395,6 @@ int codec_adapter_copy(struct comp_dev *dev)
 		return -EINVAL;
 	}
 
```

Comment:
@ranj063 for this reason this patch breaks compress play on i.MX. WIll investigate it and come back with more info.

Reply:
@dbaluta patch 2 alone will break the cadence implementation. Can you please let me know what error you're seeing with patch 1, 2 & 3?